src/m3ninx/index/segment/fst/encoding/docs/bench_test.go

```diff
@@ -0,0 +1,253 @@
+BenchmarkAccessSingleCompressedStoredFieldsData/cached_snappy-4           	  200000	      7031 ns/op	2330.03 MB/s	     128 B/op	       2 allocs/op
+BenchmarkAccessSingleCompressedStoredFieldsData/cached_deflate-4          	   10000	    137704 ns/op	 118.98 MB/s	     832 B/op	      11 allocs/op
+BenchmarkAccessSingleCompressedStoredFieldsData/cached_lz4-4              	  100000	     10149 ns/op	1614.26 MB/s	     129 B/op	       2 allocs/op
+PASS
+ok  	github.com/m3db/m3/src/m3ninx/index/segment/fst/encoding/docs	9.009s
+*/
+func BenchmarkAccessSingleCompressedStoredFieldsData(b *testing.B) {
+	pageSize := 1 << 14
+	for _, benchCase := range []struct {
+		name   string
+		cached bool
+	}{
+		{"uncached", false},
+		{"cached", true},
+	} {
+		for _, codec := range []CompressionType{SnappyCompressionType, DeflateCompressionType, LZ4CompressionType} {
+			b.Run(fmt.Sprintf("%v %v", benchCase.name, codec.String()), func(b *testing.B) {
+				var buf bytes.Buffer
+				opts := CompressionOptions{
+					Type:     codec,
+					PageSize: pageSize,
+				}
+				w, err := newCompressedDataWriter(opts, nil)
+				if err != nil {
+					panic(err.Error())
+				}
+				w.Reset(&buf)
+				offsets := make([]offset, 0, len(testDocuments))
+				for _, d := range testDocuments {
+					off, err := w.Write(d)
+					if err != nil {
+						panic(err.Error())
+					}
+					offsets = append(offsets, off)
+				}
+				if err := w.Flush(); err != nil {
+					panic(err.Error())
+				}
+				ro := ReaderOptions{
+					CompressionOptions: opts,
+				}
+				if benchCase.cached {
+					dec, err := opts.newDecompressor()
+					if err != nil {
+						panic(err.Error())
+					}
+					ro.Decompressor = dec
+				}
+				r, err := newCompressedDataReader(ro, buf.Bytes())
+				if err != nil {
+					panic(err.Error())
+				}
+
+				b.SetBytes(int64(pageSize))
+				b.ResetTimer()
+				b.ReportAllocs()
+				w.Reset(ioutil.Discard)
+				for j := 0; j < b.N; j++ {
+					i := b.N % len(testDocuments)
+					off := offsets[i]
+
+					_, err := r.Read(off)
+					if err != nil {
+						panic(err.Error())
+					}
+				}
+			})
+		}
+	}
+}
+
+/*
+ $ go test ./src/m3ninx/index/segment/fst/encoding/docs -run 'TestCompressionRatio' -v | grep -ve '\-\-' | paste -d ' ' - - | grep -v PASS |tail -n+2|cut -d':' -f2- | tr '_' ' ' | column -s ' ' -t
+
+snappy   page-size:2048     compression-ratio:  2.354
+snappy   page-size:4096     compression-ratio:  2.523
+snappy   page-size:8192     compression-ratio:  2.632
+snappy   page-size:16384    compression-ratio:  2.724
+snappy   page-size:32768    compression-ratio:  2.747
+snappy   page-size:65536    compression-ratio:  2.760
+snappy   page-size:131072   compression-ratio:  2.761
+snappy   page-size:262144   compression-ratio:  2.762
+snappy   page-size:524288   compression-ratio:  2.762
+snappy   page-size:1048576  compression-ratio:  2.762
```

Comment:
May as well just go with 4096 to mimic page size then I suppose?  Doesn't get much better the higher it gets after that really.

Reply:
Yep pretty much. The plan is to compute these compression ratios on a real world data set and see if the page size v ratio growth looks the same as synthetic data, and then pick the smallest page size we can.

Reply:
Ah yeah, hm I would think with real world the ratio gets better just since we have a lot of repeated data.